File: binary-search/binary-search.py
def binary_serach(arr, target):
    left = 0
    right = len(arr) - 1

    while left <= right:
        mid = (left + right ) // 2

        if arr[mid] == target:
            return mid
        
        elif target > arr[mid]:
            left = mid + 1

        else:
            right = mid -1


result = binary_serach([1,3,6,8,9,43,56,466,468,2003,45666], 468)
print(result)

File: binary-search/binary_search.go
package main

import (
	"fmt"
)

func BinarySerach(arr []int, target int) int {
	left := 0
	right := len(arr) - 1
	for left <= right {
		mid := (left + right) / 2
		if arr[mid] == target {
			fmt.Println("Item found at", mid)
			return mid
		} else if arr[mid] > target {
			right = mid - 1
		} else {
			left = mid + 1
		}
	}
	return -1
}

func main() {
	arr := []int{1, 3, 4, 5, 6, 66, 77, 888, 8822, 9522}
	target := 77
	searchedVal := BinarySerach(arr, target)
	fmt.Println("searched item index", searchedVal)
}
